### PR TITLE
Added e2e tests

### PR DESCRIPTION
**Pull request by zefciu**
The e2e tests are run on an actual k8s cluster. Two tests added: one
tests a blocked and one - unblocked dependency

---
This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-appcontroller/173)

### CODE REVIEW

**Comment by pigmej** [comment]
Why do we need there some custom docker images? Why they are hosted as zefciu/* ?

---

Review status: 0 of 10 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[e2e/test_basic.go, line 133 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH4pIxHCfSP77hmYSD:-KaH4pIxHCfSP77hmYSE:b-wiedm4) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/e2e/test_basic.go#L133)):*
> zefciu/sleeper

why zefciu/sleeper? How it's different from https://github.com/kubernetes/kubernetes/tree/master/build/pause ?

---

*[scripts/run_tests.sh, line 2 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH5GKg_9NppRkxO6jg:-KaH5GKhObuHUuuNsqja:b-m5cum) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/scripts/run_tests.sh#L2)):*
> zefciu/k8s-test

?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-:-KaH5JBkjUI-MQXDXLOC:b-fe7vnf)*

File: e2e/test_basic.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+	}
+	Eventually(func() bool {
+		_, depsErr := client.ResourceDefinitions().List(api.ListOptions{})
+		_, defsErr := client.Dependencies().List(api.ListOptions{})
+		return defsErr == nil && depsErr == nil
+	}, 120*time.Second, 5*time.Second).Should(BeTrue())
+	*ch <- &fixture{Namespace: ns, Controller: ac, ACClient: client}
+}
+
+var _ = Describe("Basic Suite", func() {
```
Comment (dshulyak) [inline e2e/test_basic.go]: you are missing cleanup logic after each test

File: glide.lock [hunk 2]
```diff
@@ -236,7 +312,86 @@ imports:
   - plugin/pkg/client/auth/gcp
   - plugin/pkg/client/auth/oidc
   - rest
+  - testing
   - tools/clientcmd/api
   - tools/metrics
   - transport
+- name: k8s.io/kubernetes
```
Comment (dshulyak) [inline glide.lock]: i dont see where you are using kubernetes package

**Comment by zefciu** [comment]
Done. Didn't know "pause" existed.

---

Review status: 0 of 10 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-:-KaH8DfpRe5UghGm6lij:bv1gbw4)*

File: e2e/test_basic.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	}
+	ns, err := clientset.Namespaces().Create(namespaceObj)
+	if err != nil {
+		*ch <- &fixture{Err: err}
+	}
+	appControllerObj := &v1.Pod{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "k8s-appcontroller",
+			Annotations: map[string]string{
+				"pod.alpha.kubernetes.io/init-containers": `[{"name": "kubeac-bootstrap", "image": "mirantis/k8s-appcontroller", "command": ["kubeac", "bootstrap", "/opt/kubeac/manifests"]}]`,
```
Comment (dshulyak) [inline e2e/test_basic.go]: change pull policy

File: e2e/test_basic.go [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+			},
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: "Always",
+			Containers: []v1.Container{
+				{
+					Name:    "kubeac",
+					Image:   "mirantis/k8s-appcontroller",
+					Command: []string{"/usr/bin/run_runit"},
+					Env: []v1.EnvVar{
```
Comment (dshulyak) [inline e2e/test_basic.go]: change pull policy

File: e2e/test_basic.go [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+			exErr := err.(*exec.ExitError)
+			Fail(string(out) + string(exErr.Stderr))
+		default:
+			Expect(err).NotTo(HaveOccurred())
+
+		}
+	}
+}
+
+func getFixture(ch *chan *fixture, clientset *kubernetes.Clientset) {
```
Comment (dshulyak) [inline e2e/test_basic.go]: why are u passing pointer to chan?

File: e2e/test_basic.go [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+
+	testutils "github.com/Mirantis/k8s-AppController/e2e/utils"
+	"github.com/Mirantis/k8s-AppController/pkg/client"
+	// "github.com/kr/pretty"
```
Comment (dshulyak) [inline e2e/test_basic.go]: remove

**Comment by zefciu** [comment]
Review status: 0 of 10 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[e2e/test_basic.go, line 108 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH48hv_Q3X8WUeMKTP-r1-108:-KaH_XLoogstjVSvuH1R:b-ei3rtc) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/e2e/test_basic.go#L108)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

you are missing cleanup logic after each test
</blockquote></details>

Isn't removing the namespace enough?

---

*[e2e/test_basic.go, line 133 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH4pIxHCfSP77hmYSD:-KaH_W0Naa68TZ9PKxy9:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/e2e/test_basic.go#L133)):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary><blockquote>

> zefciu/sleeper

why zefciu/sleeper? How it's different from https://github.com/kubernetes/kubernetes/tree/master/build/pause ?

</blockquote></details>

Done.

---

*[scripts/run_tests.sh, line 2 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH5GKg_9NppRkxO6jg:-KaH_VBevbpQJ42KOtBl:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/scripts/run_tests.sh#L2)):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary><blockquote>

> zefciu/k8s-test

?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by zefciu** [comment]
Review status: 0 of 11 files reviewed at latest revision, 8 unresolved discussions.

---

*[glide.lock, line 319 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH48hv_Q3X8WUeMKTR-r1-319:-KaHn9cm_lHsXjH6ih7P:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/glide.lock#L319)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

i dont see where you are using kubernetes package
</blockquote></details>

Done.

---

*[e2e/test_basic.go, line 133 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH4pIxHCfSP77hmYSD:-KaHn8J0gBLPRdm0YqH6:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/e2e/test_basic.go#L133)):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[e2e/test_basic.go, line 15 at r2](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH48hv_Q3X8WUeMKTP-r2-15:-KaHnRhwM5I-H6YpPnuT:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/68ab22b297144b6b6db3095ba0d65cabf063b3df/e2e/test_basic.go#L15)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.

---

*[e2e/test_basic.go, line 50 at r2](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH48hv_Q3X8WUeMKTP-r2-50:-KaHn1eKVXc_knplkCBb:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/68ab22b297144b6b6db3095ba0d65cabf063b3df/e2e/test_basic.go#L50)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

why are u passing pointer to chan?
</blockquote></details>

Done.

---

*[e2e/test_basic.go, line 67 at r2](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH48hv_Q3X8WUeMKTP-r2-67:-KaHn2Rccgz9KmNaE-eC:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/68ab22b297144b6b6db3095ba0d65cabf063b3df/e2e/test_basic.go#L67)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

change pull policy
</blockquote></details>

Done.

---

*[e2e/test_basic.go, line 77 at r2](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH48hv_Q3X8WUeMKTP-r2-77:-KaHnWGRafbiKjj6C8Y4:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/68ab22b297144b6b6db3095ba0d65cabf063b3df/e2e/test_basic.go#L77)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

change pull policy
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by nebril** [comment]
Reviewed 6 of 10 files at r1, 6 of 6 files at r3.
Review status: 12 of 13 files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[e2e/test_basic.go, line 49 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMvbiGT4DSJo-zXZsU:-KaMvV_pYb-cp8-uvN3V:bax5r9n) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/test_basic.go#L49)):*
> ```Go
> }
> 
> func getFixture(ch chan<- *fixture, clientset *kubernetes.Clientset) {
> ```

IMO this channel should operate on bare structs, not pointers, you are constructing all structs sent to it in this method.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-KaMxTbgxuAjANjT-Nyd:b4gpnl8) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
> ```Go
> 
> func init() {
> 	flag.StringVar(&MASTER, "master", "http://127.0.0.1:8080", "apiserver address to use with restclient")
> ```

How about using the same env var as AC binary for that? KUBERNETES_CLUSTER_URL

---

*[e2e/utils/utils.go, line 71 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMyRSYENAsRTOyGJMd:-KaMyRSYENAsRTOyGJMe:b-x6z6io) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L71)):*
> ```Go
> 	var podUpdated *v1.Pod
> 	if namespace == "" {
> 		panic("DUPA!")
> ```

Ayy lmao. Fix plz.

---

*[scripts/checkout_k8s.sh, line 7 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMzHJF3MZ_1TrLVskD:-KaMzHJF3MZ_1TrLVskE:bhcgxmu) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/scripts/checkout_k8s.sh#L7)):*
> ```Shell
> CURRENT=`pwd`
> WORKING=`mktemp -d`
> K8S_TAG=v1.5.1
> ```

K8S_TAG="${K8S_TAG:-v1.5.1}"

---

*[scripts/run_tests.sh, line 2 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMzjUKC_GfLjURiiFZ:-KaMzjUKC_GfLjURiiF_:b66lxa4) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/scripts/run_tests.sh#L2)):*
> ```Shell
> #!/bin/bash
> K8S_IMAGE="zefciu/k8s-test"
> ```

Wrong image?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by zefciu** [comment]
Review status: 12 of 13 files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[e2e/test_basic.go, line 49 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMvbiGT4DSJo-zXZsU:-KaaSzTfpp24LtnO2Z4W:b-7aknwh) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/test_basic.go#L49)):*
<details><summary><i>Previously, nebril (Maciej Kwiek) wrote…</i></summary><blockquote>

IMO this channel should operate on bare structs, not pointers, you are constructing all structs sent to it in this method.
</blockquote></details>

I wouldn't rather copy all these objects. Makes a little sense.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-KaaTCRcxLTLjb1FwmFQ:b-1arzzf) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, nebril (Maciej Kwiek) wrote…</i></summary><blockquote>

How about using the same env var as AC binary for that? KUBERNETES_CLUSTER_URL
</blockquote></details>

OK. This was copied from externalipcontroller, but I will make it consistent across our project.

---

*[e2e/utils/utils.go, line 71 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMyRSYENAsRTOyGJMd:-KaaTlBvlB85dGp6vh69:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L71)):*
<details><summary><i>Previously, nebril (Maciej Kwiek) wrote…</i></summary><blockquote>

Ayy lmao. Fix plz.
</blockquote></details>

Done.

---

*[scripts/checkout_k8s.sh, line 7 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMzHJF3MZ_1TrLVskD:-KaaTkWe3zsLnMURs2DQ:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/scripts/checkout_k8s.sh#L7)):*
<details><summary><i>Previously, nebril (Maciej Kwiek) wrote…</i></summary><blockquote>

K8S_TAG="${K8S_TAG:-v1.5.1}"
</blockquote></details>

Done.

---

*[scripts/run_tests.sh, line 2 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMzjUKC_GfLjURiiFZ:-KaaTpl30MHAyBOl3AHf:b-neygyv) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/scripts/run_tests.sh#L2)):*
<details><summary><i>Previously, nebril (Maciej Kwiek) wrote…</i></summary><blockquote>

Wrong image?
</blockquote></details>

This file is unnecessary. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by pigmej** [comment]
Review status: 10 of 12 files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-Kaax2bIvobvv8MaFawS:-Kaax2bIvobvv8MaFawT:b-vxk8v8) (no related file):*
@zefciu can we please have import.sh from master?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by zefciu** [comment]
Review status: 5 of 11 files reviewed at latest revision, 14 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-Kaax2bIvobvv8MaFawS:-KabZpkH2CMGDc00nt9M:b-8aikcl) (no related file):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary><blockquote>

@zefciu can we please have import.sh from master?
</blockquote></details>

Of course. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by nebril** [comment]
Reviewed 1 of 10 files at r1.
Review status: 5 of 11 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by pigmej** [comment]
Review status: 5 of 11 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-Kaax2bIvobvv8MaFawS:-Kafqx34-xdbfVvWesEW:b-zc6u6o) (no related file):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

Of course. Done.
</blockquote></details>

Thx :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by zefciu** [comment]
Review status: 5 of 11 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[e2e/test_basic.go, line 108 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaH48hv_Q3X8WUeMKTP-r1-108:-KafizquXsMvbnZCLTx8:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/fd3ec778b49dd087f1941204cbe2d8a9b024a1dd/e2e/test_basic.go#L108)):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

Isn't removing the namespace enough?
</blockquote></details>

Done.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-KafoCH6s_ETFi8A34w1:b-896fix) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

OK. This was copied from externalipcontroller, but I will make it consistent across our project.
</blockquote></details>

Done.

---

*[scripts/run_tests.sh, line 2 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMzjUKC_GfLjURiiFZ:-Kafj9uBy8Otwwgttcnk:bm7eclx) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/scripts/run_tests.sh#L2)):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

This file is unnecessary. 
</blockquote></details>

Done. File Removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by jellonek** [comment]
Reviewed 2 of 10 files at r1, 3 of 6 files at r3, 7 of 8 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-Kag0KutDelBxdkjDuAX:bkb1qzf) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Please use some other name, e.x. `KubernetesClusterURL` - `MASTER` seems to be meaningless.

---

*[scripts/checkout_k8s.sh, line 14 at r5](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-Kag12K4XT6M0WO0kjDS:-Kag12K4XT6M0WO0kjDT:b-mf957w) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/02c098ff158aa3c24f80dac9158ead2a4aa5ad5a/scripts/checkout_k8s.sh#L14)):*
> ```Shell
> git clone --depth 1 --single-branch https://github.com/Mirantis/kubeadm-dind-cluster.git dind
> cd $CURRENT
> echo $WORKING > $1
> ```

Missing error check on beginning of script if there is one required parameter.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by pigmej** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-Kag1YaH3PV4-3ekxZoR:b8otsxf) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Please use some other name, e.x. `KubernetesClusterURL` - `MASTER` seems to be meaningless.
</blockquote></details>

In other places we're just using `url` as a name.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by jellonek** [comment]
Reviewed 1 of 10 files at r1.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-Kag1pHKxqoAJJKtzvmP:bci8kma) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, pigmej (Jędrzej Nowak) wrote…</i></summary><blockquote>

In other places we're just using `url` as a name.
</blockquote></details>

So it would be good to reuse the same name.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by zefciu** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-Kag1zhzU36zrVPMtooH:b-cfi1ky) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

So it would be good to reuse the same name.
</blockquote></details>

This is the convention in k8s-externalipcontroller. But OK, done

---

*[scripts/checkout_k8s.sh, line 14 at r5](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-Kag12K4XT6M0WO0kjDS:-Kag2MZqAdwKhXlXjdFA:b-vno0wg) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/02c098ff158aa3c24f80dac9158ead2a4aa5ad5a/scripts/checkout_k8s.sh#L14)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Missing error check on beginning of script if there is one required parameter.
</blockquote></details>

This script is never run manually and there is no point in running it manually

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by jellonek** [comment]
Review status: 10 of 11 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[scripts/checkout_k8s.sh, line 14 at r5](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-Kag12K4XT6M0WO0kjDS:-Kag2mVgQBs87WWFLhjG:b-kctr86) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/02c098ff158aa3c24f80dac9158ead2a4aa5ad5a/scripts/checkout_k8s.sh#L14)):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

This script is never run manually and there is no point in running it manually
</blockquote></details>

Even if it is - error checking helps with catching regression in future.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by zefciu** [comment]
Review status: 10 of 11 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[scripts/checkout_k8s.sh, line 14 at r5](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-Kag12K4XT6M0WO0kjDS:-Kag3l8zaiWxGBjrOwHd:bb6un1j) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/02c098ff158aa3c24f80dac9158ead2a4aa5ad5a/scripts/checkout_k8s.sh#L14)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Even if it is - error checking helps with catching regression in future.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by jellonek** [comment]
Review status: 9 of 11 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-Kag4E5PVIGHTpxx5YcA:b-kimjl8) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, zefciu (Szymon Pyżalski) wrote…</i></summary><blockquote>

This is the convention in k8s-externalipcontroller. But OK, done
</blockquote></details>

We have:
```
cmd/bootstrap.go:       url := os.Getenv("KUBERNETES_CLUSTER_URL")
cmd/deploy.go:          url = os.Getenv("KUBERNETES_CLUSTER_URL")
cmd/get-status.go:              url = os.Getenv("KUBERNETES_CLUSTER_URL")
```

Also please remember that CamelCase is preferred in golang instead of UNDERSCORE_WITH_UPCASE.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by zefciu** [comment]
Review status: 9 of 11 files reviewed at latest revision, 8 unresolved discussions.

---

*[e2e/utils/utils.go, line 37 at r3](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-KaMxTbgxuAjANjT-Nyc:-Kag5Hx7mvMipWbuvfIZ:b-1z5gv7) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/a44f6c2d4938ab2ea6929556c8cb407a8ebc7c58/e2e/utils/utils.go#L37)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

We have:
```
cmd/bootstrap.go:       url := os.Getenv("KUBERNETES_CLUSTER_URL")
cmd/deploy.go:          url = os.Getenv("KUBERNETES_CLUSTER_URL")
cmd/get-status.go:              url = os.Getenv("KUBERNETES_CLUSTER_URL")
```

Also please remember that CamelCase is preferred in golang instead of UNDERSCORE_WITH_UPCASE.
</blockquote></details>

Done. It is "url" now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173)*

**Comment by jellonek** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r6, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-:-Kag6Q2kjg0xcp2flWdF:bnfp4nl)*

**Comment by nebril** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/173#-:-KagBebggceuylV89Fnz:bnfp4nl)*

**Pull request merged**